enlighten/SPCFileWriter.py

```diff
@@ -0,0 +1,672 @@
+# See the following refernces for details on the .spc file format implementation
+# https://github.com/bz-dev/spc-sdk
+# https://www.yumpu.com/en/document/read/40416248/a-brief-guide-to-spc-file-format-and-using-gspcio
+# https://ensembles-eu.metoffice.gov.uk/met-res/aries/technical/GSPC_UDF.PDF
+
+import os
+import sys
+import math
+import logging
+from struct import pack
+from datetime import datetime
+from dataclasses import dataclass, field
+from enum import IntEnum, IntFlag
+from tkinter import W
+
+import numpy as np
+
+log = logging.getLogger(__name__)
+
+RES_DESC_LIMIT = 9
+SRC_INSTRUMENT_LIMIT = 9
+MEMO_LIMIT = 130
+AXES_LIMIT = 30
+METHOD_FILE_LIMIT = 48
+SPARE_LIMIT = 8
+RESERVE_LIMIT = 187
+LOG_RESERVE_LIMIT = 44
+
+def fit_byte_block(field: bytearray, limit: int) -> bytes:
+    while len(field) < limit:
+        field.extend(bytearray(b"\x00"))
+    if len(field) > limit:
+        field = field[:limit]
+        field[-1] = 0 # most fields are null terminated so just make that the default
+    return bytes(field)
+
+class SPCFileType(IntFlag):
+    """
+    Describes the various file format flags.
+    See the old specification for greater detail on what each setting changes.
+    """
+    DEFAULT = 0 # Will be 32 bit single file, one trace, with even x values
```

Comment:
One general comment.  If you change # into #<! (I think) Doxygen will render per-constant API docs

Reply:
I'm not sure I completely understand. Should I change all the comments for these enums to #<! then?